src/goto-harness/goto_harness_parse_options.cpp

```diff
@@ -7,46 +7,168 @@ Author: Diffblue Ltd.
 \******************************************************************/
 
 #include <cstddef>
-#include <iostream>
+#include <set>
 #include <string>
+#include <utility>
 
+#include <goto-programs/goto_convert.h>
+#include <goto-programs/goto_model.h>
+#include <goto-programs/read_goto_binary.h>
+#include <goto-programs/write_goto_binary.h>
+#include <util/exception_utils.h>
 #include <util/exit_codes.h>
+#include <util/invariant.h>
 #include <util/version.h>
 
+#include "function_harness_generator.h"
+#include "goto_harness_generator_factory.h"
 #include "goto_harness_parse_options.h"
 
+// The basic idea is that this module is handling the following
+// sequence of events:
+// 1. Initialise a goto-model by parsing an input (goto) binary
+// 2. Initialise the harness generator (with some config) that will handle
+//    the mutation of the goto-model.
+// 3. Write the end result of that process to the output binary
+
```

Comment:
Can we add  a brief description of the harness generation here? In particular, to say that a new function is created that represents the harness, and that one needs to make it the entry point (via `cbmc --function`) in order to use it.

Reply:
Good point; Arguably this information should be added to the user manual as well